functions/Get-DbaSsisEnvironment.ps1

```diff
@@ -0,0 +1,229 @@
+
+.EXAMPLE
+Get-DbaSsisEnvironment -SqlInstance localhost -Environment DEV -Folder DWH_ETL
+
+Gets variables of 'DEV' environment located in 'DWH_ETL' folder on 'localhost' Server
+
+.EXAMPLE
+Get-DbaSsisEnvironment -SqlInstance localhost -Environment DEV -Folder DWH_ETL, DEV2, QA
+
+Gets variables of 'DEV' environment(s) located in folders 'DWH_ETL', 'DEV2' and 'QA' on 'localhost' server
+
+.EXAMPLE
+Get-DbaSsisEnvironment -SqlInstance localhost -Environment DEV -FolderExclude DWH_ETL, DEV2, QA
+
+Gets variables of 'DEV' environments located in folders other than 'DWH_ETL', 'DEV2' and 'QA' on 'localhost' server
+
+.EXAMPLE
+Get-DbaSsisEnvironment -SqlInstance localhost -Environment DEV, PROD -Folder DWH_ETL, DEV2, QA
+
+Gets variables of 'DEV' and 'PROD' environment(s) located in folders 'DWH_ETL', 'DEV2' and 'QA' on 'localhost' server
+
+.EXAMPLE
+Get-DbaSsisEnvironment -SqlInstance localhost -EnvironmentExclude DEV, PROD -Folder DWH_ETL, DEV2, QA
+
+Gets variables of environments other than 'DEV' and 'PROD' located in folders 'DWH_ETL', 'DEV2' and 'QA' on 'localhost' server
+
+.EXAMPLE
+Get-DbaSsisEnvironment -SqlInstance localhost -EnvironmentExclude DEV, PROD -FolderEcxclude DWH_ETL, DEV2, QA
+
+Gets variables of environments other than 'DEV' and 'PROD' located in folders other than 'DWH_ETL', 'DEV2' and 'QA' on 'localhost' server
+
+.NOTES
+Author: Bartosz Ratajczyk ( @b_ratajczyk )
+
+dbatools PowerShell module (https://dbatools.io)
+Copyright (C) 2016 Chrissy LeMaire
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program. If not, see http://www.gnu.org/licenses/.
+#>
+function Get-DbaSsisEnvironment {
+
+[CmdletBinding()]
+	Param (
+		[parameter(Mandatory, ValueFromPipeline)]
+		[Alias('SqlServer', 'ServerInstance')]
+		[DbaInstanceParameter[]]$SqlInstance,
+        [Parameter(Mandatory=$false)]
+        [System.Management.Automation.PSCredential]$SqlCredential,
+        [parameter(Mandatory=$false)]
+		[object[]]$Environment,
+        [parameter(Mandatory=$false)]
+        [object[]]$EnvironmentExclude,
+        [parameter(Mandatory=$false)]
+        [object[]]$Folder,
+        [parameter(Mandatory=$false)]
+        [object[]]$FolderExclude,
+		[switch]$Silent
+	)
+
+    begin {
+
+    }
+
+    process {
+
+        foreach($instance in $SqlInstance)
+        {
+            try
+            {
+                Write-Message -Message "Connecting to $instance" -Level 5 -Silent $Silent
```

Comment:
You can remove the `-Silent $Silent` from all the Write-Message and Stop-Function calls. This will get picked up by default if `-Silent` is passed in by the calling function.